app/controllers/api/physical_storages_controller.rb

```diff
@@ -20,6 +20,28 @@ def create_resource(_type, _id = nil, data = {})
       action_result(false, err.to_s)
     end
 
+    def delete_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        begin
+          physical_storage = resource_search(id, type, klass)
+          unless physical_storage.supports?(:delete)
+            error_msg = "#{physical_storage.name} raised the following error: " + physical_storage.unsupported_reason(:delete)
```

Comment:
@agrare I've done all the requested modifications. The only issue left is my suggestion for error_msg. if you prefer to keep it simple I will change it to
`raise physical_storage.unsupported_reason(:delete) unless physical_storage.supports?(:delete)`

Reply:
You should use string interpolation instead of `+` so `"#{physical_storage.name} raised the following error: #{physical_storage.unsupported_reason(:delete)}"`

Reply:
I'm actually surprised rubocop didn't complain about this cc @bdunne

Reply:
> You should use string interpolation instead of `+` so `"#{physical_storage.name} raised the following error: #{physical_storage.unsupported_reason(:delete)}"`

fixed. I changed the error msg a bit to: `error_msg = "Failed to delete #{physical_storage.name}: #{physical_storage.unsupported_reason(:delete)}"`